crates/flux-fhir-analysis/src/wf/sortck.rs

```diff
@@ -216,11 +220,27 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
         }
     }
 
+    fn synth_lit(&mut self, lit: fhir::Lit, fhir_id: FhirId, span: Span) -> rty::Sort {
```

Comment:
Maybe pass the orignal `expr` instead of the `fhir_id` and `span`

```suggestion
    fn synth_lit(&mut self, lit: fhir::Lit, expr: &fhir::Expr) -> rty::Sort {
```